todo/src/App.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import logo from './logo.svg';
+import './App.css';
+import { connect } from 'react-redux';
+import { addTodo, toggleComplete, deleteTodo } from './actions';
+import ListWrapper from './styles/listWrapper';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      newTodo: ''
+    }
+  }
+
+  componentDidMount() {
+    if(localStorage.getItem('todos') !== null && localStorage.getItem('todos') !== 'null'){
+      let myTodos = JSON.parse(localStorage.getItem('todos'));
+      myTodos.forEach((todo,index) => {
+        this.props.addTodo(todo.todo);
+        if(todo.completed){
+          this.props.toggleComplete(index);
+        }
+      })
+    }
+  }
+
+  componentDidUpdate() {
+    localStorage.setItem('todos',JSON.stringify(this.props.todos))
+  }
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
+  addTheTodo = ev => {
+    ev.preventDefault();
+    this.props.addTodo(this.state.newTodo);
+    this.setState({newTodo: ''})
+  }
+
+  toggleComplete = index => {
+    this.props.toggleComplete(index);
+    console.log('fired toggle')
+  }
+
+  deleteTodo = (event,index) => {
+    event.stopPropagation();
+    this.props.deleteTodo(index);
+  }
+
+  render() {
+    return (
+      <ListWrapper>
+        <h1 className='list-title'>My Todo List</h1>
```

Comment:
It is better to decouple them into different components so it will be reusable if need in anywhere else, I know in this case it won't be needed but it is good practice to get used to.

[Container Components](https://medium.com/@learnreact/container-components-c0e67432e005)